tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -3130,6 +3141,7 @@ bool SPIRVEmitter::emitEntryFunctionWrapper(const FunctionDecl *decl,
   // Construct the wrapper function signature.
   const uint32_t voidType = theBuilder.getVoidType();
   const uint32_t funcType = theBuilder.getFunctionType(voidType, {});
+  entryFunctionDecl = const_cast<FunctionDecl*>(decl);
```

Comment:
I am thinking about whether we should keep `AddExecutionModeForEntryPoint()` as a method and keep a dedicated class field solely for it. It seems to me that by embedding the contents of `AddExeuctionModeForEntryPoint()` in where it is called should be fine.